### ISSUE TITLE
Force TargetFramework override in CSPROJ?

### ISSUE THREAD

**Issue by asocialraccoon** [dotnet/sdk#14561]
Is there a way to change a project's target framework if it has been specified as a build parameter or passed down via `ProjectReference`'s  `SetTargetFramework` property?

When running `dotnet publish <...> -r win-x64 -f net5.0` any `<TargetFramework>` properties seem to be completely ignored. Looking to do something like this to force Windows builds to use a specific Windows-specific TFM:
```
<TargetFramework Condition="'$(TargetFramework)' == 'net5.0' AND $(RuntimeIdentifier.StartsWith('win'))">net5.0-windows10.0.17763.0</TargetFramework>
```

**Comment by wli3** [comment]
It is by design commandline will override the property from the project. Is there any reason you need to only build windows tfm on windows?

**Comment by asocialraccoon** [comment]
> Is there any reason you need to only build windows tfm on windows?

Yes, I would like to enforce that due to various functionality in the project being reliant on WinRT. From my understanding .NET 5.0 doesn't provide any other means of linking those calls other than targeting a supporting framework. This was previously not an issue with .NET Core since I could package reference `Microsoft.Windows.SDK.Contracts` and it would just work where applicable - it looks like this was completely disregarded as a convenience along the way?

> It is by design commandline will override the property from the project.

No, but it would be nice to be able to do this. As I pointed out this also applies when a project is referenced - I'm not sure if a `net5.0` application can interop with say a `net5.0-windows` dependency but assuming it is possible, being able to "upgrade" the referenced project's framework would be a welcome option.
